extra_data/write_cxi.py

```diff
@@ -185,44 +265,49 @@ def write(self, filename, fillvalues=None):
 
         layouts = self.collect_data()
 
+        data_label = self.image_label
         _fillvalues = {
-            # data can be uint16 (raw) or float32 (proc)
-            'data': np.nan if layouts['data'].dtype.kind == 'f' else 0,
+            # Data can be uint16 (raw) or float32 (proc)
+            data_label: np.nan if layouts[data_label].dtype.kind == 'f' else 0,
             'gain': 0,
             'mask': 0xffffffff
         }
         if fillvalues:
             _fillvalues.update(fillvalues)
-        # enforce that fill values are compatible with array dtype
-        _fillvalues['data'] = layouts['data'].dtype.type(_fillvalues['data'])
+        # Enforce that fill values are compatible with array dtype
+        _fillvalues[data_label] = layouts[data_label].dtype.type(
+            _fillvalues[data_label])
         if 'gain' in layouts:
-            _fillvalues['gain'] = layouts['gain'].dtype.type(_fillvalues['gain'])
+            _fillvalues['gain'] = layouts['gain'].dtype.type(
+                _fillvalues['gain'])
         if 'mask' in layouts:
-            _fillvalues['mask'] = layouts['mask'].dtype.type(_fillvalues['mask'])
+            _fillvalues['mask'] = layouts['mask'].dtype.type(
+                _fillvalues['mask'])
 
         log.info("Writing to %s", filename)
 
-        # Virtual datasets require HDF5 >= 1.10. Specifying this up front should
-        # mean it fails before touching the file if run on an older version.
-        # We also specify this as the maximum version, to ensure we're creating
-        # files that can be read by HDF5 1.10.
+        # Virtual datasets require HDF5 >= 1.10.
+        # Specifying this up front should mean it fails before touching
+        # the file if run on an older version. We also specify this as
+        # the maximum version, to ensure we're creating files that can
+        # be read by HDF5 1.10.
         with h5py.File(filename, 'w', libver=('v110', 'v110')) as f:
             f.create_dataset('cxi_version', data=[150])
             d = f.create_dataset('entry_1/experiment_identifier',
                                  shape=experiment_ids.shape,
                                  dtype=h5py.special_dtype(vlen=str))
             d[:] = experiment_ids
 
-            # pulseId, trainId, cellId are not part of the CXI standard, but
-            # it allows extra data.
+            # pulseId, trainId, cellId are not part of the CXI standard,
+            # but it allows extra data.
             f.create_dataset('entry_1/pulseId', data=pulse_ids)
```

Comment:
I think we should avoid creating a `pulseId` dataset for JUNGFRAU, given that it's not even claiming to store pulse IDs. This bit isn't part of the CXI format (the format is fairly flexible about adding in extra datasets).

Reply:
Renamed a `pulseId` dataset for JUNGFRAU to `memoryCell`.

Reply:
OK. This will mean, if I'm following correctly, that it has a regular `memoryCell` dataset plus a virtual `cellId` one which have the same data when you read them? That seems a little bit confusing, but not a big problem.

Reply:
Yes, right now it produces `memoryCell` with the shape [n_frames] and `cellId` with the same data with the shape [n_frames x n_modules]. Maybe we should just drop `cellId` for JUNGFRAU? As far as I remember, we either way require `memoryCell` to be the same for each module...

Reply:
Aha, I had missed the difference in their shape.

I *think* it's reasonable to check that the memory cell numbers match up across modules, but this kind of depends on the details of how the detector works. For *pulse* IDs, this is an important check because we want to be sure that a single frame from the virtual CXI file is all coming from the same pulse. In theory, different modules could store the same pulse in different memory cells, but I think the detectors are always configured so all modules act in the same way, in which case this is a useful sanity check.